apps/intellij/src/main/kotlin/dev/nx/console/toolWindow/NxProjectsTreeStructure.kt

```diff
@@ -0,0 +1,327 @@
+        TreeUtil.installActions(tree)
+        installPopupActions()
+    }
+
+    override fun getRootElement(): Any = root
+
+    fun updateNxProjects(nxWorkspace: NxWorkspace?) {
+        this.nxWorkspace = nxWorkspace
+        root = NxSimpleNode.Root(nxWorkspace)
+        treeModel.invalidate()
+    }
+
+    sealed class NxSimpleNode(parent: SimpleNode?) : CachingSimpleNode(parent) {
+        class Root(val nxWorkspace: NxWorkspace?) : NxSimpleNode(null) {
+
+            init {
+                icon = NxIcons.Action
+            }
+
+            override fun buildChildren(): Array<SimpleNode> {
+                if (nxWorkspace == null) {
+                    return emptyArray()
+                }
+                val tasks: Array<SimpleNode> = arrayOf(Tasks(nxWorkspace, this))
+                val apps: Array<SimpleNode> = arrayOf(Apps(nxWorkspace, this))
+                val libs: Array<SimpleNode> = arrayOf(Libs(nxWorkspace, this))
+                return tasks + apps + libs
+            }
+
+            // TODO do not add a root node or get the name of workspace properly?
+            override fun getName(): String =
+                nxWorkspace?.workspacePath?.substringAfterLast("/") ?: "nx"
+        }
+
+        class Apps(private val nxWorkspace: NxWorkspace, parent: SimpleNode) : NxSimpleNode(null) {
+
+            init {
+                icon = AllIcons.Nodes.ModuleGroup
+            }
+
+            override fun buildChildren(): Array<SimpleNode> =
+                nxWorkspace.workspace.projects.values
+                    .filter { it.projectType == "application" }
+                    .map { Project(it, this) }
+                    .toTypedArray()
+
+            override fun getName(): String = "apps"
+        }
+
+        class Libs(private val nxWorkspace: NxWorkspace, parent: SimpleNode) : NxSimpleNode(null) {
+
+            init {
+                icon = AllIcons.Nodes.PpLibFolder
+            }
+
+            override fun buildChildren(): Array<SimpleNode> =
+                nxWorkspace.workspace.projects.values
+                    .filter { it.projectType == "application" }
+                    .map { Project(it, this) }
+                    .toTypedArray()
+
+            override fun getName(): String = "libs"
+        }
+
+        class Tasks(private val nxWorkspace: NxWorkspace, parent: SimpleNode) :
+            NxSimpleNode(parent) {
+            init {
+                icon = AllIcons.Nodes.ConfigFolder
+            }
+
+            override fun buildChildren(): Array<SimpleNode> =
+                nxWorkspace.workspace.projects.values
+                    .flatMap { p -> p.targets.keys.map { it to p.name } }
+                    .groupBy { it.first }
+                    .map { Task(nxWorkspace, it.key, this) }
+                    .toTypedArray()
+
+            override fun getName(): String = "Tasks"
+        }
+
+        class Task(private val nxWorkspace: NxWorkspace, val taskName: String, parent: SimpleNode) :
```

Comment:
I would name these nodes 'targets' instead of tasks. They are not the same thing - a target is the definition in your `project.json` and is used to create a task which is a single invocation of a target. Here is a list of the targets defined in all the projects so it should be called targets.

In the future we will need nodes for configurations too but it doesn't have to be in this PR.

Reply:
ok for naming targets

Yeah I tried the run configuration but doesn't work well...